recipes/libsigcpp/2.x.x/conanfile.py

```diff
@@ -0,0 +1,111 @@
+import os
+
+from conans.errors import ConanInvalidConfiguration
+from conans import ConanFile, Meson, tools
+
+
+class LibSigCppConan(ConanFile):
+    name = "libsigcpp"
+    homepage = "https://github.com/libsigcplusplus/libsigcplusplus"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "LGPL-3.0"
+    description = (
+        "libsigc++ implements a typesafe callback system for standard C++."
+    )
+    topics = ("conan", "libsigcpp", "callback")
+    settings = "os", "compiler", "arch", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {"shared": False, "fPIC": True}
+
+    _meson = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def build_requirements(self):
+        self.build_requires("meson/0.56.0")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
```

Comment:
```suggestion
            del self.options.fPIC
        if self.settings.compiler == "Visual Studio":
            self.options.shared = True
```

I don't like to see the default profile broken when installing a package. There are other few cases where we use shared by default, but mostly because only shared is accepted for any OS.

Reply:
I was still looking into this, but I cannot reproduce this problem on Windows. Building libsigcpp directly (statically and shared), it builds as it should (including the right library suffixes etc.). For building the package, I am stuck at Conan not being able to find my VS 15 (or 16) installation.

It would be great though if we can short circuite this for now this way!